### DISCUSSION TITLE
How to add `Premain-Class` attribute to JAR manifest?

### DISCUSSION THREAD

**Discussion by amordahl** [com-lihaoyi/mill#3134] [Q&A]
I am new to using Mill, and I am trying to build an instrumentation agent, but I can't figure out how to generate a manifest file that includes a `Premain-Class` attribute. I noticed [issue 624](https://github.com/com-lihaoyi/mill/issues/624) said that "You can overwrite the `def manifest: T[JarManifest]` target and append any manifest entry you want" but I'm not really sure how to achieve this. Thanks!

**Comment by lefou** [comment]
You can add a `Premain-Class` entry to the manifest by overriding the `manifest` target of your `JavaModule` or `ScalaModule`:

```scala
def manifest = T {
  super.manifest().add("Premain-Class" -> "org.example.PremainAgent")
}
```
